src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenCrossVersionTest.groovy

```diff
@@ -0,0 +1,47 @@
+package hudson.plugins.gradle.injection
+
+
+import hudson.plugins.gradle.BuildScanAction
+import hudson.plugins.gradle.BuildScanBuildWrapper
+import hudson.tasks.Maven
+import org.jvnet.hudson.test.CreateFileBuilder
+import org.jvnet.hudson.test.JenkinsRule
+import org.jvnet.hudson.test.ToolInstallations
+import spock.lang.Unroll
+
+@Unroll
+class BuildScanInjectionMavenCrossVersionTest extends BaseInjectionIntegrationTest {
+
+    def 'build scan is discovered from Maven build - #mavenVersion'(String mavenVersion) {
+        given:
+        withGlobalEnvVars {
+            put('JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_INJECTION', 'true')
+            put('JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_EXTENSION_VERSION', '1.14.2')
+            put('JENKINSGRADLEPLUGIN_CCUD_EXTENSION_VERSION', '1.10.1')
+        }
+
+        def p = j.createFreeStyleProject()
+        p.buildWrappersList.add(new BuildScanBuildWrapper())
+        p.buildersList.add(new CreateFileBuilder('pom.xml', MavenExamples.simplePom()))
+        p.buildersList.add(new CreateFileBuilder('.mvn/gradle-enterprise.xml', MavenExamples.gradleEnterpriseConfiguration()))
+        def mavenInstallation = ToolInstallations.configureDefaultMaven("apache-maven-${mavenVersion}", Maven.MavenInstallation.MAVEN_30)
+        p.buildersList.add(new Maven('package', mavenInstallation.name, null, '', '', false, null, null))
+
+        def slave = createSlave('foo')
+        p.setAssignedNode(slave)
+
+        when:
+        def build = j.buildAndAssertSuccess(p)
+
+        then:
+        println JenkinsRule.getLog(build)
+        if (mavenVersion >= '3.3.1') {
+            def action = build.getAction(BuildScanAction)
+            action.scanUrls.size() == 1
+            new URL(action.scanUrls.get(0))
+        }
```

Comment:
As far as I understand this branch will be added after `1.15.3` release. Please add a comment/todo to update it, similar to the one in `build.gradle`.

Reply:
No, this works only for Maven > 3.3.1. We aren't publishing anything for Maven < 3.3.1 so the only thing we will test is that the build succeeds. Actually, we can just test that no url is published.

Reply:
The warning message for the older versions, should we see it in the logs?

Reply:
Only for 3.2.5, however I would not tests such details here as they are already tested.

Reply:
The current check LGTM 👍